spellbook/models/x2y2/ethereum/x2y2_ethereum_events.sql

```diff
@@ -0,0 +1,180 @@
+    FROM {{ source('x2y2_ethereum','X2Y2_r1_evt_EvProfit') }} prof
+    INNER JOIN {{ source('x2y2_ethereum','X2Y2_r1_evt_EvInventory') }} inv ON inv.itemHash = prof.itemHash
+    LEFT JOIN {{ ref('tokens_ethereum_nft') }} tokens ON ('0x' || substring(get_json_object(inv.item, '$.data'), 155, 40)) = tokens.contract_address
+    LEFT JOIN {{ ref('nft_ethereum_aggregators') }} agg ON agg.contract_address=taker
+    WHERE taker IN (SELECT contract_address FROM {{ ref('nft_ethereum_aggregators') }})
+    )
+
+, direct_x2y2_txs AS (
+    SELECT inv.evt_block_time AS block_time
+    , CASE WHEN inv.currency = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' THEN maker ELSE taker END AS buyer
+    , prof.to AS seller
+    , ROUND(bytea2numeric_v2(substring(get_json_object(inv.item, '$.data'), 195,64)),0) AS token_id
+    , get_json_object(inv.item, '$.price') AS amount_raw
+    , prof.currency AS currency_contract
+    , '0x' || substring(get_json_object(inv.item, '$.data'), 155, 40) AS project_contract_address
+    , tokens.name AS collection
+    , NULL AS aggregator_name
+    , NULL AS aggregator_address
+    , inv.evt_tx_hash AS tx_hash
+    , prof.evt_index
+    , COALESCE(get_json_object(inv.item, '$.price')*get_json_object(get_json_object(inv.detail, '$.fees[0]'), '$.percentage')/1e6, 0) AS platform_fee_amount_raw
+    , COALESCE(get_json_object(get_json_object(inv.detail, '$.fees[0]'), '$.percentage')/1e6, 0) AS platform_fee_percentage
+    , COALESCE(get_json_object(inv.item, '$.price')*get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.percentage')/1e6, 0) AS royalty_fee_amount_raw
+    , COALESCE(get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.percentage')/1e6, 0) AS royalty_fee_percentage
+    , get_json_object(get_json_object(inv.detail, '$.fees[1]'), '$.to') AS royalty_fee_receive_address
+    FROM  {{ source('x2y2_ethereum','X2Y2_r1_evt_EvProfit') }} prof 
+    INNER JOIN {{ source('x2y2_ethereum','X2Y2_r1_evt_EvInventory') }} inv ON inv.itemHash=prof.itemHash
+    LEFT JOIN {{ ref('tokens_ethereum_nft') }} tokens ON ('0x' || substring(get_json_object(inv.item, '$.data'), 155, 40)) = tokens.contract_address
+    WHERE taker NOT IN (SELECT contract_address FROM {{ ref('nft_ethereum_aggregators') }})
+    )
+
+, aggregator_routed_x2y2_txs_formatted AS (
+    SELECT block_time
+    , e721.to AS buyer
+    , seller
+    , token_id
+    , amount_raw
+    , currency_contract
+    , project_contract_address
+    , collection
+    , 'Bundle Trade' AS trade_type
+    , aggregator_name
+    , aggregator_address
+    , tx_hash
+    , txs.evt_index
+    , platform_fee_amount_raw
+    , platform_fee_percentage
+    , royalty_fee_amount_raw
+    , royalty_fee_percentage
+    , royalty_fee_receive_address
+    FROM aggregator_routed_x2y2_txs txs
+    LEFT JOIN {{ source('erc721_ethereum','evt_transfer') }} e721 ON txs.tx_hash = e721.evt_tx_hash
+        AND txs.token_id = e721.tokenId
```

Comment:
You can also add an incremental filter here after the LEFT JOIN:
{% if is_incremental() %}
-- this filter will only be applied on an incremental run
AND e721.evt_block_time >= (select max(block_time) from {{ this }})
{% endif %}